src/Data/Nat/Logarithm.agda

```diff
@@ -0,0 +1,54 @@
+-----------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Logarithm base 2 and respective properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Nat.Logarithm where
+
+open import Data.Nat 
+open import Data.Nat.Induction using (<-wellFounded)
+open import Data.Nat.Logarithm.Core
+open import Relation.Binary.PropositionalEquality using (_≡_)
+
+------------------------------------------------------------------------
+-- Logarithm base 2
+
+-- Floor version
+
+⌊log₂_⌋ : ℕ → ℕ
+⌊log₂ n ⌋ = ⌊log2⌋ n (<-wellFounded n)
+
+-- Ceil version
+
+⌈log₂_⌉ : ℕ → ℕ
+⌈log₂ n ⌉ = ⌈log2⌉ n (<-wellFounded n)
+
+------------------------------------------------------------------------
+-- Properties of ⌊log₂⌋ and ⌊log₂_⌋
+
+⌊log₂⌊n/2⌋⌋≡⌊log₂n⌋∸1 : ∀ {n} → ⌊log₂ ⌊ n /2⌋ ⌋ ≡ ⌊log₂ n ⌋ ∸ 1
+⌊log₂⌊n/2⌋⌋≡⌊log₂n⌋∸1 {n} = ⌊log2⌋⌊n/2⌋≡⌊log2⌋n∸1 {n}
+
+⌊log₂2*b⌋≡1+⌊log₂b⌋ : ∀ b → b > 0 → ⌊log₂ (2 * b) ⌋ ≡ 1 + ⌊log₂ b ⌋
```

Comment:
Still a rogue `b` and also I think there should be brackets around the `2*n`, i.e. `⌊log₂[2*n]⌋≡1+⌊log₂n⌋`? Same for many proofs below.

We could also use the new `NonZero` instance argument scheme for this (and similar) proofs?, e.g.
```
⌊log₂[2*n]⌋≡1+⌊log₂n⌋ : ∀ n .{{_ : NonZero n}} → ⌊log₂ (2 * n) ⌋ ≡ 1 + ⌊log₂ n ⌋
```

Reply:
Definitely! At the time I was not using the most recent std library version but I will do that!

Reply:
Done!